e2eTests/scenarios/IndirectTemplate/runtest.ps1

```diff
@@ -0,0 +1,244 @@
+# |_____|_| |_|\__,_|_|_|  \___|\___|\__|    |_|\___|_| |_| |_| .__/|_|\__,_|\__\___|
+#                                                             | |
+#                                                             |_|
+# This test tests the following scenario:
+#
+#  - Create a new repository based on the PTE template with no apps (this will be the "indirect" template repository)
+#  - Create a new repository based on the PTE template with 1 app, using compilerfolder and donotpublishapps (this will be the "final" template repository)
+#  - Run Update AL-Go System Files in final repo (using indirect repo as template)
+#  - Run Update AL-Go System files in indirect repo
+
+# TODO: describe the scenario
+
+
+#  - Validate that custom step is present in indirect repo
+#  - Run Update AL-Go System files in final repo
+#  - Validate that custom step is present in final repo
+#  - Run Update AL-Go System files in final repo
+#  - Validate that both custom steps is present in final repo
+#
+'@
+
+$errorActionPreference = "Stop"; $ProgressPreference = "SilentlyContinue"; Set-StrictMode -Version 2.0
+$prevLocation = Get-Location
+
+Remove-Module e2eTestHelper -ErrorAction SilentlyContinue
+Import-Module (Join-Path $PSScriptRoot "..\..\e2eTestHelper.psm1") -DisableNameChecking
+. (Join-Path -Path $PSScriptRoot -ChildPath '..\..\..\Actions\CheckForUpdates\yamlclass.ps1')
+. (Join-Path -Path $PSScriptRoot -ChildPath "..\..\..\Actions\CheckForUpdates\CheckForUpdates.HelperFunctions.ps1")
+
+$templateRepository = "$githubOwner/$repoName-template"
+$repository = "$githubOwner/$repoName"
+$branch = "main"
+
+$template = "https://github.com/$pteTemplate"
+
+# Login
+SetTokenAndRepository -github:$github -githubOwner $githubOwner -token $token -repository $repository
+
+# Create tempolate repository
+CreateAlGoRepository `
+    -github:$github `
+    -linux:$linux `
+    -template $template `
+    -repository $templateRepository `
+    -branch $branch
+$templateRepoPath = (Get-Location).Path
+
+Set-Location $prevLocation
+
+$appName = 'MyApp'
+$publisherName = 'Contoso'
+
+# Create final repository
+CreateAlGoRepository `
+    -github:$github `
+    -linux:$linux `
+    -template $template `
+    -repository $repository `
+    -branch $branch `
+    -contentScript {
+        Param([string] $path)
+        $null = CreateNewAppInFolder -folder $path -name $appName -publisher $publisherName
+    }
+$finalRepoPath = (Get-Location).Path
+
+# Update AL-Go System Files to use template repository
+RunUpdateAlGoSystemFiles -directCommit -wait -templateUrl $templateRepository -ghTokenWorkflow $token -repository $repository -branch $branch | Out-Null
+
+Set-Location $templateRepoPath
+
+Pull
+
+# Make modifications to the template repository
+
+# Add Custom Jobs to CICD.yaml
+$cicdWorkflow = Join-Path $templateRepoPath '.github/workflows/CICD.yaml'
+$cicdYaml = [yaml]::Load($cicdWorkflow)
+$cicdYaml | Should -Not -BeNullOrEmpty
+
+# Modify the permissions
```

Comment:
Doesn't look like the E2E test modifies the permissions. I think that's a scenario we should test if we support it